### PR TITLE
Add Docker Image and CI/CD

### PR DESCRIPTION
**Pull request by glrf**

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to quay.io
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_IO_USERNAME }}
+          password: ${{ secrets.QUAY_IO_PASSWORD }}
```
Comment (ccremer) [inline .github/workflows/push.yml:35]: We could also use GitHub's registry
```suggestion
      - name: Login to ghcr.io
        uses: docker/login-action@v1
        with:
          registry: ghcr.io
          username: ${{ github.repository_owner }}
          password: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        run: echo "VERSION=latest" >> ${GITHUB_ENV}
+      - name: Set image version from tag
+        if: startsWith(github.ref, 'refs/tags/v')
+        run: echo "VERSION=$(echo ${GITHUB_REF#refs/tags/})" >> ${GITHUB_ENV}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```
Comment (ccremer) [inline .github/workflows/push.yml:28]: Why do we need QEMU and Buildx if we're only pushing 1 architecture (linux/amd64)?

**Review by simu — APPROVED** [review]
LGTM

File: .github/workflows/push.yml (hunk 2 above)
Comment (glrf) [reply .github/workflows/push.yml:28]: Good point. Copy :spaghetti: error

File: .github/workflows/push.yml (hunk 1 above)
Comment (glrf) [reply .github/workflows/push.yml:35]: I don't have a preference. So yes why not.

**Pull request merged**